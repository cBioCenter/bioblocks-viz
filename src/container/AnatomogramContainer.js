"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
// tslint:disable-next-line:import-name
var anatomogram_1 = require("anatomogram");
var immutable_1 = require("immutable");
var React = require("react");
var react_redux_1 = require("react-redux");
var redux_1 = require("redux");
var action_1 = require("~bioblocks-viz~/action");
var component_1 = require("~bioblocks-viz~/component");
var container_1 = require("~bioblocks-viz~/container");
var data_1 = require("~bioblocks-viz~/data");
var helper_1 = require("~bioblocks-viz~/helper");
var reducer_1 = require("~bioblocks-viz~/reducer");
var selector_1 = require("~bioblocks-viz~/selector");
var AnatomogramContainerClass = /** @class */ (function (_super) {
    tslib_1.__extends(AnatomogramContainerClass, _super);
    function AnatomogramContainerClass(props) {
        var _this = _super.call(this, props) || this;
        _this.divRef = null;
        _this.onClick = function (ids) {
            // Anatomogram returns an array of strings for click events, but we only ever work with a single id.
            var id = ids[0];
            var _a = _this.props, addLabel = _a.onLabelAdd, removeLabel = _a.onLabelRemove, selectIds = _a.selectIds;
            if (selectIds.includes(id)) {
                removeLabel(id);
            }
            else {
                addLabel(id);
            }
        };
        _this.deriveIdsFromSpecies = function (species) { return Object.keys(data_1.AnatomogramMapping[species]); };
        _this.onMouseOut = function (id) {
            return;
        };
        _this.onMouseOver = function (id) {
            return;
        };
        _this.parseCategory = function (category) {
            var splitCategories = category.split(/-|_/);
            return splitCategories[0];
        };
        _this.resizeSVGElement = function (error, svgDomNode) {
            if (_this.divRef) {
                var isSvgHeightBigger = svgDomNode.height.baseVal.value > svgDomNode.width.baseVal.value;
                // The Anatomogram Component internally sets the svg height to 'auto'.
                // So, to allow more flexibility in sizing it, we have to manually override it here. Sorry.
                svgDomNode.style.height = isSvgHeightBigger ? "calc(" + _this.divRef.style.height + " - 50px)" : 'auto';
                svgDomNode.style.padding = '0';
                svgDomNode.style.width = isSvgHeightBigger ? 'auto' : "calc(" + _this.divRef.style.width + " - 75px)";
            }
        };
        _this.state = {
            ids: _this.deriveIdsFromSpecies(props.species),
        };
        return _this;
    }
    AnatomogramContainerClass.prototype.setupDataServices = function () {
        this.registerDataset('cells', []);
        this.registerDataset('labels', []);
        reducer_1.BioblocksMiddlewareTransformer.addTransform(this.getCellToLabelTransform());
        reducer_1.BioblocksMiddlewareTransformer.addTransform(this.getLabelToCellTransform());
    };
    AnatomogramContainerClass.prototype.componentDidUpdate = function (prevProps) {
        var species = this.props.species;
        if (species !== prevProps.species) {
            this.setState({
                ids: this.deriveIdsFromSpecies(species),
            });
        }
    };
    AnatomogramContainerClass.prototype.render = function () {
        var _this = this;
        var _a = this.props, iconSrc = _a.iconSrc, species = _a.species, selectIds = _a.selectIds;
        var ids = this.state.ids;
        return (React.createElement(react_redux_1.Provider, { store: reducer_1.BBStore },
            React.createElement("div", { className: 'anatomogram-container', ref: function (node) {
                    if (node) {
                        _this.divRef = node.getElementsByTagName('div')[0];
                    }
                } },
                React.createElement(component_1.ComponentCard, { componentName: 'Anatomogram', iconSrc: iconSrc },
                    React.createElement(anatomogram_1.default, { atlasUrl: "", highlightColour: 'yellow', onClick: this.onClick, onInjected: this.resizeSVGElement, onMouseOut: this.onMouseOut, onMouseOver: this.onMouseOver, selectColour: 'ffaa00', selectIds: selectIds.toArray(), showIds: ids, species: species, selectedView: species === 'mus_musculus' ? 'female' : 'male' })))));
    };
    AnatomogramContainerClass.prototype.getCellToLabelTransform = function () {
        var _this = this;
        return {
            fn: function (state) {
                var e_1, _a, e_2, _b, e_3, _c;
                var species = _this.props.species;
                var currentCells = selector_1.selectCurrentItems(state, 'cells').toArray();
                var _d = selector_1.getSpring(state), category = _d.category, graphData = _d.graphData;
                var result = immutable_1.Set();
                try {
                    for (var currentCells_1 = tslib_1.__values(currentCells), currentCells_1_1 = currentCells_1.next(); !currentCells_1_1.done; currentCells_1_1 = currentCells_1.next()) {
                        var cellIndex = currentCells_1_1.value;
                        // Parse labels and categories from SPRING data.
                        var labelForCategory = graphData.nodes[cellIndex] ? graphData.nodes[cellIndex].labelForCategory : {};
                        var labels = data_1.AnatomogramMapping[species][labelForCategory[category]];
                        if (labels) {
                            labels.forEach(function (label) { return (result = result.add(label)); });
                        }
                        try {
                            for (var _e = (e_2 = void 0, tslib_1.__values(Object.keys(data_1.AnatomogramMapping[species]))), _f = _e.next(); !_f.done; _f = _e.next()) {
                                var id = _f.value;
                                try {
                                    for (var _g = (e_3 = void 0, tslib_1.__values(Object.values(labelForCategory))), _h = _g.next(); !_h.done; _h = _g.next()) {
                                        var label = _h.value;
                                        if (data_1.AnatomogramMapping[species][id].includes(label)) {
                                            result = result.add(id);
                                        }
                                    }
                                }
                                catch (e_3_1) { e_3 = { error: e_3_1 }; }
                                finally {
                                    try {
                                        if (_h && !_h.done && (_c = _g.return)) _c.call(_g);
                                    }
                                    finally { if (e_3) throw e_3.error; }
                                }
                            }
                        }
                        catch (e_2_1) { e_2 = { error: e_2_1 }; }
                        finally {
                            try {
                                if (_f && !_f.done && (_b = _e.return)) _b.call(_e);
                            }
                            finally { if (e_2) throw e_2.error; }
                        }
                    }
                }
                catch (e_1_1) { e_1 = { error: e_1_1 }; }
                finally {
                    try {
                        if (currentCells_1_1 && !currentCells_1_1.done && (_a = currentCells_1.return)) _a.call(currentCells_1);
                    }
                    finally { if (e_1) throw e_1.error; }
                }
                return result;
            },
            fromState: { stateName: 'cells' },
            toState: { stateName: 'labels' },
        };
    };
    AnatomogramContainerClass.prototype.getLabelToCellTransform = function () {
        var _this = this;
        return {
            fn: function (state) {
                var _a = _this.props, species = _a.species, selectIds = _a.selectIds;
                var graphData = selector_1.getSpring(state).graphData;
                var anatomogramMap = data_1.AnatomogramMapping[species];
                var candidateLabels = immutable_1.Set();
                selectIds.forEach(function (id) {
                    candidateLabels = id && anatomogramMap[id] ? candidateLabels.merge(anatomogramMap[id]) : candidateLabels;
                });
                var cellIndices = immutable_1.Set();
                graphData.nodes.forEach(function (node) {
                    var labelsForNode = Object.values(node.labelForCategory);
                    candidateLabels.forEach(function (label) {
                        if (label && labelsForNode.includes(label)) {
                            cellIndices = cellIndices.add(node.number);
                            return;
                        }
                    });
                });
                return cellIndices;
            },
            fromState: 'bioblocks/labels',
            toState: 'bioblocks/cells',
        };
    };
    AnatomogramContainerClass.defaultProps = {
        iconSrc: '',
        onLabelAdd: helper_1.EMPTY_FUNCTION,
        onLabelRemove: helper_1.EMPTY_FUNCTION,
        selectIds: immutable_1.Set(),
        species: 'homo_sapiens',
    };
    AnatomogramContainerClass.displayName = 'Anatomogram';
    return AnatomogramContainerClass;
}(container_1.BioblocksVisualization));
exports.AnatomogramContainerClass = AnatomogramContainerClass;
var mapStateToProps = function (state) { return ({
    selectIds: selector_1.selectCurrentItems(state, 'labels'),
}); };
var mapDispatchToProps = function (dispatch) {
    return redux_1.bindActionCreators({
        onLabelAdd: action_1.createContainerActions('labels').add,
        onLabelRemove: action_1.createContainerActions('labels').remove,
    }, dispatch);
};
exports.AnatomogramContainer = react_redux_1.connect(mapStateToProps, mapDispatchToProps)(AnatomogramContainerClass);
//# sourceMappingURL=AnatomogramContainer.js.map