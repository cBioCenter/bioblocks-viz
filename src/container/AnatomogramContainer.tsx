// tslint:disable-next-line:import-name
import Anatomogram from 'anatomogram';
import { isEqual } from 'lodash';
import * as React from 'react';

import {
  CellContext,
  ICellContext,
  initialCellContext,
  initialSpringContext,
  ISpringContext,
  SpringContext,
} from '~chell-viz~/context';
import { CHELL_CSS_STYLE } from '~chell-viz~/data';

export interface IAnatomogramContainerProps {
  cellContext: ICellContext;
  height: number | string;
  species: 'homo_sapiens' | 'mus_musculus';
  springContext: ISpringContext;
  style: CHELL_CSS_STYLE;
  width: number | string;
}

export interface IAnatomogramContainerState {
  ids: string[];
  selectIds: string[];
}

export const springToAnatomogramMapping: { [key: string]: { [key: string]: string } } = {
  homo_sapiens: {
    P11A: 'UBERON_0000178',
    P11B: 'UBERON_0000473',
    P12A: 'UBERON_0000955',
    P9A: 'UBERON_0000977',
  },
  mus_musculus: {
    bladder: 'UBERON_0001255',
    heart: 'UBERON_0000948',
    kidney: 'UBERON_0002113',
    liver: 'UBERON_0002107',
    lung: 'UBERON_0002048 ',
    mammary: 'UBERON_0001911',
    marrow: 'UBERON_0002371',
    muscle: 'sUBERON_0014892',
    spleen: 'UBERON_0002106',
    thymus: 'UBERON_0002370',
    tongue: 'UBERON_0001723',
    trachea: 'UBERON_0003126',
  },
};

export const anatomogramToSpringMapping: { [key: string]: { [key: string]: string } } = {
  homo_sapiens: {
    UBERON_0000178: 'P11A',
    UBERON_0000473: 'P11B',
    UBERON_0000955: 'P12A',
    UBERON_0000977: 'P9A',
  },
  mus_musculus: {
    UBERON_0000948: 'heart',
    UBERON_0001255: 'bladder',
    UBERON_0001723: 'tongue',
    UBERON_0001911: 'mammary',
    UBERON_0002048: 'lung',
    UBERON_0002106: 'spleen',
    UBERON_0002107: 'liver',
    UBERON_0002113: 'kidney',
    UBERON_0002370: 'thymus',
    UBERON_0002371: 'marrow',
    UBERON_0003126: 'trachea',
    sUBERON_0014892: 'muscle',
  },
};

export class AnatomogramContainerClass extends React.Component<IAnatomogramContainerProps, IAnatomogramContainerState> {
  public static defaultProps = {
    cellContext: {
      ...initialCellContext,
    },
    height: '300px',
    ids: ['UBERON_0000178', 'UBERON_0000473', 'UBERON_0000955', 'UBERON_0000977'],
    springContext: {
      ...initialSpringContext,
    },
    style: {},
    width: '400px',
  };

  public static displayName = 'Anatomogram';

  protected anatomogramRef: any;

  constructor(props: IAnatomogramContainerProps) {
    super(props);
    this.state = {
      ids: Object.keys(anatomogramToSpringMapping[props.species]),
      selectIds: [],
    };
  }

  public componentDidUpdate(prevProps: IAnatomogramContainerProps) {
    const { cellContext, springContext } = this.props;
    if (!isEqual(springContext.selectedCategories, prevProps.springContext.selectedCategories)) {
      this.setState({
        selectIds: springContext.selectedCategories.map(
          category => springToAnatomogramMapping[this.props.species][category],
        ),
      });
    } else if (!isEqual(cellContext.currentCells, prevProps.cellContext.currentCells)) {
      const categories = new Set<string>();
      for (const cellIndex of cellContext.currentCells) {
        categories.add(springContext.graphData.nodes[cellIndex].category);
      }
      this.setState({
        selectIds: Array.from(categories).map(category => springToAnatomogramMapping[this.props.species][category]),
      });
    }
  }

  public render() {
    const { ids, selectIds } = this.state;

    return (
      <div className={'anatomogram-container'} style={{ height: '100%' }}>
        <Anatomogram
          atlasUrl={``}
          highlightColour={'yellow'}
          onClick={this.onClick}
          onMouseOut={this.onMouseOut}
          onMouseOver={this.onMouseOver}
          selectColour={'ffaa00'}
          selectIds={selectIds}
          showIds={ids}
          species={this.props.species}
          selectedView={this.props.species === 'mus_musculus' ? 'female' : 'male'}
        />
      </div>
    );
  }

  protected onClick = (id: string) => {
    this.props.springContext.toggleCategory(anatomogramToSpringMapping[this.props.species][id]);
  };

  protected onMouseOut = (id: string) => {
    return;
  };

  protected onMouseOver = (id: string) => {
    return;
  };
}

type requiredProps = Omit<IAnatomogramContainerProps, keyof typeof AnatomogramContainerClass.defaultProps> &
  Partial<IAnatomogramContainerProps>;

export const AnatomogramContainer = (props: requiredProps) => (
  <CellContext.Consumer>
    {cellContext => (
      <SpringContext.Consumer>
        {springContext => (
          <AnatomogramContainerClass {...props} cellContext={cellContext} springContext={springContext} />
        )}
      </SpringContext.Consumer>
    )}
  </CellContext.Consumer>
);
