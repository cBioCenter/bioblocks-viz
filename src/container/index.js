"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
tslib_1.__exportStar(require("./BioblocksVisualization"), exports);
tslib_1.__exportStar(require("./AnatomogramContainer"), exports);
tslib_1.__exportStar(require("./InfoPanelContainer"), exports);
tslib_1.__exportStar(require("./NGLContainer"), exports);
tslib_1.__exportStar(require("./ContactMapContainer"), exports);
tslib_1.__exportStar(require("./ProteinFeatureViewer"), exports);
tslib_1.__exportStar(require("./SpringContainer"), exports);
tslib_1.__exportStar(require("./TensorTContainer"), exports);
tslib_1.__exportStar(require("./UMAPSequenceContainer"), exports);
tslib_1.__exportStar(require("./UMAPTranscriptionalContainer"), exports);
//# sourceMappingURL=index.js.map