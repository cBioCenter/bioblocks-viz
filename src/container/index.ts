export * from './BioblocksVisualization';

export * from './AnatomogramContainer';
export * from './InfoPanelContainer';
export * from './NGLContainer';
export * from './ContactMapContainer';
export * from './ProteinFeatureViewer';
export * from './SpringContainer';
export * from './TensorTContainer';
export * from './UMAPSequenceContainer';
export * from './UMAPTranscriptionalContainer';
