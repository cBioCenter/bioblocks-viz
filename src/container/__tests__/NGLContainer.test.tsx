import { mount, shallow } from 'enzyme';
import * as React from 'react';
import { Provider } from 'react-redux';

import { Checkbox, Icon, Popup, Table } from 'semantic-ui-react';
import { NGLContainer, NGLContainerClass } from '~bioblocks-viz~/container';
import { BioblocksPDB } from '~bioblocks-viz~/data';
import { Store } from '~bioblocks-viz~/reducer';

describe('NGLContainer', () => {
  it('Should match the default snapshot when hooked up to a redux store.', () => {
    const store = Store;
    const wrapper = mount(
      <Provider store={store}>
        <NGLContainer />
      </Provider>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('Should match the default snapshot when not hooked up to a redux store.', () => {
    const wrapper = shallow(<NGLContainerClass />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Should handle experimental PDB files.', async () => {
    const pdbs = await Promise.all([
      BioblocksPDB.createPDB('exp_1_sample.pdb'),
      BioblocksPDB.createPDB('exp_2_sample.pdb'),
    ]);

    const wrapper = shallow(<NGLContainerClass experimentalProteins={pdbs} />);

    expect(wrapper.instance().state).toEqual({
      selectedExperimentalProteins: ['exp_1_sample'],
      selectedPredictedProteins: [],
    });
    wrapper.setProps({
      experimentalProteins: [pdbs[1]],
    });
    wrapper.update();
    expect(wrapper.instance().state).toEqual({
      selectedExperimentalProteins: ['exp_2_sample'],
      selectedPredictedProteins: [],
    });
  });

  it('Should handle predicted PDB files.', async () => {
    const pdbs = await Promise.all([
      BioblocksPDB.createPDB('pred_1_sample.pdb'),
      BioblocksPDB.createPDB('pred_2_sample.pdb'),
    ]);

    const wrapper = shallow(<NGLContainerClass predictedProteins={pdbs} />);

    expect(wrapper.instance().state).toEqual({
      selectedExperimentalProteins: [],
      selectedPredictedProteins: ['pred_1_sample'],
    });
    wrapper.setProps({
      predictedProteins: [pdbs[1]],
    });
    wrapper.update();
    expect(wrapper.instance().state).toEqual({
      selectedExperimentalProteins: [],
      selectedPredictedProteins: ['pred_2_sample'],
    });
  });

  it('Should handle selecting PDB files via checkbox.', async () => {
    const pdbs = await Promise.all([
      BioblocksPDB.createPDB('exp_1_sample.pdb'),
      BioblocksPDB.createPDB('exp_2_sample.pdb'),
      BioblocksPDB.createPDB('pred_1_sample.pdb'),
      BioblocksPDB.createPDB('pred_2_sample.pdb'),
    ]);

    const wrapper = mount(
      <NGLContainerClass experimentalProteins={[pdbs[0], pdbs[1]]} predictedProteins={[pdbs[2], pdbs[3]]} />,
    );

    expect(wrapper.instance().state).toEqual({
      selectedExperimentalProteins: ['exp_1_sample'],
      selectedPredictedProteins: ['pred_1_sample'],
    });

    wrapper
      .find(Popup)
      .at(0)
      .simulate('click');
    wrapper
      .find(Checkbox)
      .at(1)
      .simulate('change');
    expect(wrapper.instance().state).toEqual({
      selectedExperimentalProteins: ['exp_1_sample', 'exp_2_sample'],
      selectedPredictedProteins: ['pred_1_sample'],
    });
    wrapper
      .find(Checkbox)
      .at(0)
      .simulate('change');
    expect(wrapper.instance().state).toEqual({
      selectedExperimentalProteins: ['exp_2_sample'],
      selectedPredictedProteins: ['pred_1_sample'],
    });

    wrapper
      .find(Icon)
      .at(1)
      .simulate('click');
    wrapper
      .find(Checkbox)
      .at(3)
      .simulate('change');
    expect(wrapper.instance().state).toEqual({
      selectedExperimentalProteins: ['exp_2_sample'],
      selectedPredictedProteins: ['pred_1_sample', 'pred_2_sample'],
    });
    wrapper
      .find(Checkbox)
      .at(2)
      .simulate('change');
    expect(wrapper.instance().state).toEqual({
      selectedExperimentalProteins: ['exp_2_sample'],
      selectedPredictedProteins: ['pred_2_sample'],
    });
  });

  it('Should not show a popup when there are no PDB files to select.', async () => {
    const wrapper = mount(<NGLContainerClass />);

    wrapper
      .find(Popup)
      .at(0)
      .simulate('click');
    expect(wrapper.find(Checkbox).length).toBe(0);

    wrapper
      .find(Icon)
      .at(1)
      .simulate('click');
    expect(wrapper.find(Checkbox).length).toBe(0);

    const pdbs = await Promise.all([
      BioblocksPDB.createPDB('exp_1_sample.pdb'),
      BioblocksPDB.createPDB('exp_2_sample.pdb'),
    ]);

    wrapper.setProps({ experimentalProteins: pdbs, predictedProteins: pdbs });
    wrapper
      .find(Popup)
      .at(0)
      .simulate('click');
    expect(wrapper.find(Checkbox).length).not.toBe(0);

    wrapper
      .find(Icon)
      .at(1)
      .simulate('click');
    expect(wrapper.find(Checkbox).length).not.toBe(0);
  });

  it('Should handle clearing PDB files.', async () => {
    const pdbs = await Promise.all([
      BioblocksPDB.createPDB('pred_1_sample.pdb'),
      BioblocksPDB.createPDB('pred_2_sample.pdb'),
    ]);

    const wrapper = mount(<NGLContainerClass experimentalProteins={pdbs} predictedProteins={pdbs} />);
    wrapper
      .find(Icon)
      .at(0)
      .simulate('click');
    expect(
      wrapper
        .find(Table.Cell)
        .at(1)
        .text(),
    ).toEqual('Experimental (1/2)');
    expect(
      wrapper
        .find(Table.Cell)
        .at(2)
        .text(),
    ).toEqual('Predicted (1/2)');

    wrapper.setProps({
      experimentalProteins: [],
      predictedProteins: [],
    });
    wrapper.update();
    expect(
      wrapper
        .find(Table.Cell)
        .at(1)
        .text(),
    ).toEqual('Experimental (0/0)');
    expect(
      wrapper
        .find(Table.Cell)
        .at(2)
        .text(),
    ).toEqual('Predicted (0/0)');
  });
});
