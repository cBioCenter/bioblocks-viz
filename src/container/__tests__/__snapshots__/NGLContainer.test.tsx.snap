// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NGLContainer Should match the default snapshot when hooked up to a redux store. 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(NGLContainerClass)>
    <NGLContainerClass
      addCandidateResidues={[Function]}
      addHoveredResidues={[Function]}
      addLockedResiduePair={[Function]}
      candidateResidues={Array []}
      dispatchNglFetch={[Function]}
      experimentalProteins={Array []}
      hoveredResidues={Array []}
      hoveredSecondaryStructures={Array []}
      isDataLoading={false}
      lockedResiduePairs={Immutable.Map {}}
      maxPDBPerPopup={5}
      measuredProximity="C-α"
      predictedProteins={Array []}
      removeAllLockedResiduePairs={[Function]}
      removeCandidateResidues={[Function]}
      removeHoveredResidues={[Function]}
      removeLockedResiduePair={[Function]}
      removeNonLockedResidues={[Function]}
      selectedSecondaryStructures={Array []}
      showConfigurations={true}
    >
      <Grid
        padded={true}
      >
        <div
          className="ui padded grid"
        >
          <GridRow
            centered={true}
            stretched={true}
          >
            <div
              className="centered stretched row"
            >
              <Popup
                disabled={false}
                keepInViewPort={true}
                on="click"
                position="bottom center"
                trigger={
                  <Icon
                    as="i"
                    name="tasks"
                    size="large"
                  />
                }
                wide={true}
              >
                <Portal
                  closeOnDocumentClick={true}
                  closeOnEscape={true}
                  closeOnTriggerClick={true}
                  eventPool="default"
                  onClose={[Function]}
                  onMount={[Function]}
                  onOpen={[Function]}
                  onUnmount={[Function]}
                  openOnTriggerClick={true}
                  trigger={
                    <Icon
                      as="i"
                      name="tasks"
                      size="large"
                    />
                  }
                  triggerRef={[Function]}
                >
                  <Ref
                    innerRef={[Function]}
                  >
                    <RefFindNode
                      innerRef={[Function]}
                    >
                      <Icon
                        as="i"
                        name="tasks"
                        onBlur={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onMouseEnter={[Function]}
                        onMouseLeave={[Function]}
                        size="large"
                      >
                        <i
                          aria-hidden="true"
                          className="tasks large icon"
                          onBlur={[Function]}
                          onClick={[Function]}
                          onFocus={[Function]}
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                        />
                      </Icon>
                    </RefFindNode>
                  </Ref>
                </Portal>
              </Popup>
            </div>
          </GridRow>
          <GridRow>
            <div
              className="row"
            >
              <NGLComponent
                addCandidateResidues={[Function]}
                addHoveredResidues={[Function]}
                addLockedResiduePair={[Function]}
                backgroundColor="#ffffff"
                candidateResidues={Array []}
                dispatchNglFetch={[Function]}
                experimentalProteins={Array []}
                height="90%"
                hoveredResidues={Array []}
                hoveredSecondaryStructures={Array []}
                isDataLoading={false}
                lockedResiduePairs={Object {}}
                maxPDBPerPopup={5}
                measuredProximity="C-α"
                onMeasuredProximityChange={[Function]}
                onResize={[Function]}
                predictedProteins={Array []}
                removeAllLockedResiduePairs={[Function]}
                removeCandidateResidues={[Function]}
                removeHoveredResidues={[Function]}
                removeLockedResiduePair={[Function]}
                removeNonLockedResidues={[Function]}
                selectedSecondaryStructures={Array []}
                showConfigurations={true}
                width="100%"
              >
                <ComponentCard
                  componentName="NGL Viewer"
                  frameHeight={0}
                  frameWidth={0}
                  headerHeight={32}
                  height="525px"
                  iconSrc="assets/icons/spring-icon.png"
                  isFramedComponent={false}
                  isFullPage={false}
                  padding={0}
                  showSettings={false}
                  width="525px"
                >
                  <Card
                    centered={true}
                    className="bioblocks-component-card"
                    style={
                      Object {
                        "height": "525px",
                        "maxWidth": "unset",
                        "padding": "0 0 5px 5px",
                        "width": "525px",
                      }
                    }
                  >
                    <div
                      className="ui centered card bioblocks-component-card"
                      onClick={[Function]}
                      style={
                        Object {
                          "height": "525px",
                          "maxWidth": "unset",
                          "padding": "0 0 5px 5px",
                          "width": "525px",
                        }
                      }
                    >
                      <Menu
                        secondary={true}
                        style={
                          Object {
                            "height": 32,
                            "margin": 0,
                          }
                        }
                      >
                        <div
                          className="ui secondary menu"
                          style={
                            Object {
                              "height": 32,
                              "margin": 0,
                            }
                          }
                        >
                          <MenuItem
                            fitted="horizontally"
                            position="left"
                            style={
                              Object {
                                "margin": 0,
                              }
                            }
                          >
                            <div
                              className="left horizontally fitted item"
                              onClick={[Function]}
                              style={
                                Object {
                                  "margin": 0,
                                }
                              }
                            >
                              <img
                                alt="component icon"
                                src="assets/icons/spring-icon.png"
                                style={
                                  Object {
                                    "height": "32px",
                                    "padding": "2px",
                                    "width": "32px",
                                  }
                                }
                              />
                              NGL Viewer
                            </div>
                          </MenuItem>
                          <MenuItem
                            fitted="horizontally"
                            position="right"
                            style={
                              Object {
                                "margin": 0,
                              }
                            }
                          >
                            <div
                              className="right horizontally fitted item"
                              onClick={[Function]}
                              style={
                                Object {
                                  "margin": 0,
                                }
                              }
                            >
                              <Icon
                                as="i"
                                name="expand arrows alternate"
                                onClick={[Function]}
                              >
                                <i
                                  aria-hidden="true"
                                  className="expand arrows alternate icon"
                                  onClick={[Function]}
                                />
                              </Icon>
                            </div>
                          </MenuItem>
                        </div>
                      </Menu>
                      <div
                        className="NGLComponent"
                        style={
                          Object {
                            "height": "90%",
                            "width": "100%",
                          }
                        }
                      >
                        <Dimmer
                          active={false}
                        >
                          <DimmerInner
                            active={false}
                          >
                            <Ref
                              innerRef={
                                Object {
                                  "current": <div
                                    class="ui dimmer"
                                  >
                                    <div
                                      class="content"
                                    >
                                      <div
                                        class="ui loader"
                                      />
                                    </div>
                                  </div>,
                                }
                              }
                            >
                              <RefFindNode
                                innerRef={
                                  Object {
                                    "current": <div
                                      class="ui dimmer"
                                    >
                                      <div
                                        class="content"
                                      >
                                        <div
                                          class="ui loader"
                                        />
                                      </div>
                                    </div>,
                                  }
                                }
                              >
                                <div
                                  className="ui dimmer"
                                  onClick={[Function]}
                                >
                                  <div
                                    className="content"
                                  >
                                    <Loader>
                                      <div
                                        className="ui loader"
                                      />
                                    </Loader>
                                  </div>
                                </div>
                              </RefFindNode>
                            </Ref>
                          </DimmerInner>
                        </Dimmer>
                        <div
                          className="NGLCanvas"
                          onKeyDown={[Function]}
                          onMouseLeave={[Function]}
                          role="img"
                          style={
                            Object {
                              "height": "100%",
                              "width": "100%",
                            }
                          }
                        />
                      </div>
                      <div
                        style={
                          Object {
                            "bottom": "27px",
                            "height": "27px",
                            "position": "relative",
                            "right": "5px",
                          }
                        }
                      >
                        <Button
                          active={false}
                          as="button"
                          floated="right"
                          onClick={[Function]}
                          toggle={true}
                        >
                          <Ref
                            innerRef={
                              Object {
                                "current": <button
                                  aria-pressed="false"
                                  class="ui toggle right floated button"
                                >
                                  Superposition: off
                                </button>,
                              }
                            }
                          >
                            <RefFindNode
                              innerRef={
                                Object {
                                  "current": <button
                                    aria-pressed="false"
                                    class="ui toggle right floated button"
                                  >
                                    Superposition: off
                                  </button>,
                                }
                              }
                            >
                              <button
                                aria-pressed={false}
                                className="ui toggle right floated button"
                                onClick={[Function]}
                              >
                                Superposition: off
                              </button>
                            </RefFindNode>
                          </Ref>
                        </Button>
                      </div>
                    </div>
                  </Card>
                </ComponentCard>
              </NGLComponent>
            </div>
          </GridRow>
        </div>
      </Grid>
    </NGLContainerClass>
  </Connect(NGLContainerClass)>
</Provider>
`;

exports[`NGLContainer Should match the default snapshot when not hooked up to a redux store. 1`] = `
<Grid
  padded={true}
>
  <GridRow
    centered={true}
    stretched={true}
  >
    <Popup
      disabled={false}
      keepInViewPort={true}
      on="click"
      position="bottom center"
      trigger={
        <Icon
          as="i"
          name="tasks"
          size="large"
        />
      }
      wide={true}
    >
      <Table
        as="table"
        basic="very"
        style={
          Object {
            "maxWidth": "100%",
          }
        }
      >
        <TableHeader
          as="thead"
        >
          <TableRow
            as="tr"
            cellAs="td"
          >
            <TableHeaderCell
              as="th"
            >
              Select Structure to View
            </TableHeaderCell>
          </TableRow>
        </TableHeader>
        <TableHeader
          as="thead"
        >
          <TableRow
            as="tr"
            cellAs="td"
          >
            <TableCell
              as="td"
            >
              Experimental (0/0)
            </TableCell>
            <TableCell
              as="td"
            >
              Predicted (0/0)
            </TableCell>
          </TableRow>
        </TableHeader>
        <TableBody
          as="tbody"
        >
          <TableRow
            as="tr"
            cellAs="td"
          >
            <TableCell
              as="td"
              width={3}
            >
              <Form
                as="form"
                style={
                  Object {
                    "height": "165px",
                    "overflow": "auto",
                  }
                }
              />
            </TableCell>
            <TableCell
              as="td"
              width={1}
            >
              <Form
                as="form"
                style={
                  Object {
                    "height": "165px",
                    "overflow": "auto",
                  }
                }
              />
            </TableCell>
          </TableRow>
        </TableBody>
      </Table>
    </Popup>
  </GridRow>
  <GridRow>
    <NGLComponent
      addCandidateResidues={[Function]}
      addHoveredResidues={[Function]}
      addLockedResiduePair={[Function]}
      backgroundColor="#ffffff"
      candidateResidues={Array []}
      dispatchNglFetch={[Function]}
      experimentalProteins={Array []}
      height="90%"
      hoveredResidues={Array []}
      hoveredSecondaryStructures={Array []}
      isDataLoading={false}
      lockedResiduePairs={Object {}}
      maxPDBPerPopup={5}
      measuredProximity="C-α"
      onMeasuredProximityChange={[Function]}
      onResize={[Function]}
      predictedProteins={Array []}
      removeAllLockedResiduePairs={[Function]}
      removeCandidateResidues={[Function]}
      removeHoveredResidues={[Function]}
      removeLockedResiduePair={[Function]}
      removeNonLockedResidues={[Function]}
      selectedSecondaryStructures={Array []}
      showConfigurations={true}
      width="100%"
    />
  </GridRow>
</Grid>
`;
